src/java.base/windows/classes/sun/nio/fs/WindowsUserDefinedFileAttributeView.java

```diff
@@ -51,8 +51,17 @@ private String join(String file, String name) {
             throw new NullPointerException("'name' is null");
         return file + ":" + name;
     }
+
     private String join(WindowsPath file, String name) throws WindowsException {
-        return join(file.getPathForWin32Calls(), name);
+        if (name == null)
+            throw new NullPointerException("'name' is null");
+        WindowsFileSystem wfs = file.getFileSystem();
+        WindowsPath namePath = WindowsPath.parse(wfs, name);
+        if (namePath.isAbsolute())
+            throw new IllegalArgumentException("'name' is not relative");
+        String path = join(file.getPathForWin32Calls(), name);
+        WindowsPath wp = WindowsPath.createFromNormalizedPath(wfs, path);
+        return wp.getPathForWin32Calls();
```

Comment:
I think we can simplify this to avoid most of the conversions, I'll get back to you soon with a proposal for this.

Reply:
One thing I'm still mulling over here is whether there should be a check that namePath only has one name element.  That is, I think "join" should probably reject the name if it has a root component or has more than one name element. That would replace the check for an absolute path because that would imply a root component.